web/src/engine/websites/MangaTown.ts

```diff
@@ -2,23 +2,22 @@ import { Tags } from '../Tags';
 import icon from './MangaTown.webp';
 import { type Chapter, DecoratableMangaScraper, Page } from '../providers/MangaPlugin';
 import * as Common from './decorators/Common';
+import { FetchWindowScript } from '../platform/FetchProvider';
 
 //Modded DM5 script
 const pageScript = `
     new Promise(async (resolve, reject) => {
         try {
             const pagecount = window.total_pages;
-            const images = await Promise.all([...new Array(pagecount).keys()].map(async p => {
-                eval(await $.ajax({
-                    url: 'chapterfun.ashx',
-                    data: {
-                        cid: window.chapter_id,
-                        page: p + 1,
-                    }
-                }));
-                return d[0];
-            }));
-            resolve(images.map(link => new URL(link, window.location.href).href));
+            const pages = [];
+            for (let page = 1; page <= pagecount; page++) {
+                const response = await fetch('chapterfun.ashx?cid='+ window.chapter_id + '&page=' + page);
+                const script = await response.text();
+                eval(script);
+                pages.push(d[0]);
```

Comment:
May compose FQDN here instead of mapping later `new URL(d[0], window.location.href).href)`